### PR TITLE
fix: Workaround ActualHeight bug for ShadowContainer

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes https://github.com/unoplatform/uno/issues/16049

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:
- [ ] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Tested the changes where applicable:
	- [ ] UWP
	- [ ] WinUI
	- [ ] iOS
	- [ ] Android
	- [ ] WASM
	- [ ] MacOS
- [ ] Updated the documentation as needed:
	- [ ] [General Doc Update](https://github.com/unoplatform/uno.toolkit.ui/tree/main/doc)
	- [ ] [Controls Doc Update](https://github.com/unoplatform/uno.toolkit.ui/tree/main/doc/controls)
	- [ ] [Extensions Doc Update](https://github.com/unoplatform/uno.toolkit.ui/tree/main/doc/helpers)
	- [ ] [controls-styles.md](https://github.com/unoplatform/uno.toolkit.ui/blob/main/doc/controls-styles.md)
	- [ ] [lightweight-styling.md (LightWeight Styling Resource Keys)](https://github.com/unoplatform/uno.toolkit.ui/blob/main/doc/lightweight-styling.md)
- [ ] [Runtime Tests and/or UI Tests](https://platform.uno/docs/articles/contributing/guidelines/creating-tests.html) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal)
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.Toolkit.Skia.WinUI/Controls/Shadows/ShadowContainer.cs [hunk 1]
```diff
@@ -114,6 +114,16 @@ void OnContentChanged(DependencyObject sender, DependencyProperty dp)
...
+
+			// This is not necessary on WinUI, but is necessary on Uno due to behavior mismatch in ActualWidth/ActualHeight.
+			// For the ShadowContainerSamplePage, when we add a new shadow, we are expanding the StackPanel that's wrapped in ShadowContainer.
+			// We get through InvalidateCanvasLayout via OnContentSizeChanged and we update SKXamlCanvas.Height to the new Height
+			// On WinUI, the updated Height is immediately reflected on ActualHeight, but that's not the case on Uno.
+			// So, SKXamlCanvas still has its internal SKCanvas with the old size (as it relies on ActualHeight)
+			// see https://github.com/mono/SkiaSharp/blob/a290ccffc74bc95a45569e8b74e77be851d4a688/source/SkiaSharp.Views.Uno/SkiaSharp.Views.Uno.WinUI.Shared/SKXamlCanvas.cs#L140
+			// Then we end up drawing the old size.
+			// To account for Uno's mismatched behavior, we call InvalidateShadows.
+			InvalidateShadows();
```
Comment (agneszitte) [inline src/Uno.Toolkit.Skia.WinUI/Controls/Shadows/ShadowContainer.cs:126]: @Youssef1313 can we add some tests for this?

**Review by Xiaoy312 — APPROVED** [review]

File: src/Uno.Toolkit.Skia.WinUI/Controls/Shadows/ShadowContainer.cs (hunk 1 above)
Comment (Youssef1313) [reply src/Uno.Toolkit.Skia.WinUI/Controls/Shadows/ShadowContainer.cs:126]: I think we are not running tests in CI against Skia anyway. It's not reproducible on Wasm (and possible all the other platforms)

Comment (agneszitte) [reply src/Uno.Toolkit.Skia.WinUI/Controls/Shadows/ShadowContainer.cs:126]: cc @kazo0

**Comment by Youssef1313** [comment]
@kazo0 iOS is failing to build. It's unrelated to the change done here. Is it a known issue?

**Comment by kazo0** [comment]
> @kazo0 iOS is failing to build. It's unrelated to the change done here. Is it a known issue?

@Youssef1313 smells like the default xcode version was changed for the macos agents

**Comment by kazo0** [comment]
Should hopefully be fixed by https://github.com/unoplatform/uno.toolkit.ui/pull/1093

**Comment by agneszitte** [comment]
@Youssef1313, @kazo0, @Xiaoy312  the PR branch will need to be rebased with the main branch to pick up the change for https://github.com/unoplatform/uno.toolkit.ui/pull/1093

File: src/Uno.Toolkit.Skia.WinUI/Controls/Shadows/ShadowContainer.cs (hunk 1 above)
Comment (kazo0) [reply src/Uno.Toolkit.Skia.WinUI/Controls/Shadows/ShadowContainer.cs:126]: We should add support for Skia on the CI (https://github.com/unoplatform/uno.toolkit.ui/issues/1095)

Is there still anything here that could be covered by a test? Even if it's not yet on the CI?

**Review by Xiaoy312 — APPROVED** [review]
could use some comments to explain whats going on
and, how a 50x100 panel in the shadow-container with the two shadows adds up to 160px height?
100base + (10offset + 20blur) * 2both-sides

**Comment by kazo0** [comment]
Will need to be rebased again once https://github.com/unoplatform/uno.toolkit.ui/pull/1098 merges 🙃

**Comment by kazo0** [comment]
@Youssef1313 the build should be good now once you rebase with latest `main`

**Review by kazo0 — APPROVED** [review]

**Pull request merged**